server/README.md

```diff
@@ -0,0 +1,472 @@
+>   to another write in the object storage at the same location.
+> - The object storage follows an eventual consistency strategy, hence the new block upload
+>   is guaranteed to overwrite the previous one.
+> - The only issue would be if multiple attempts of the same block ID would be done with
+>   different data on each try. In this case eventual consistency means it is possible
+>   for a short period of time to get a previous value when querying the block.
+>   However this is purely theoretical since block ID is random (hence there should be
+>   no concurrency between clients) and a client considers a block as immutable.
+
+### Invitations
+
+During the process of invitation, one of the client (the claimer) doesn't have a device
+yet (as it is the whole point of the invitation process !).
+Hence using certificates during this step is not possible (since a certificate is signed by a device).
+
+No certificate involvement means:
+
+- Invitations data are totally separated from other data (given other data always relies
+  on certificates to do part of there validation).
+- Invitation request doesn't contains a client-generated timestamp.
+
+This last point is great from a causality point of view:
+
+- With certificates the client propose a change that occurred at a given timestamp, then it's
+  up to the server to check for causality to know if it should accept it.
+- Here the client merely ask the server do to something, then the server checks if it is
+  possible *now*, then do it *now*. No causality can be broken given we only work in the present.
+
+Regarding concurrency, locks are needed to protect the creation of invitation against:
+
+- Concurrent invitation creation and user creation leading an invitation for an existing user.
+  This is handled by simply taking the `common` topic lock.
+- Concurrent invitation creations of the same email creating multiple invitations
+  (see [example in Data concurrency paragraph](#data-concurrency)). This is handled
+  by a dedicated `invitation_creation` advisory lock that must be taken before the
+  invitation creation procedure starts any checks involving the invitations.
+
+### What is locked by who ?
+
+The following array recapitulate which topics (and other related locks) each command
+needs to lock (and in which mode).
+
+Topic locks: `common`, `realm`, `sequester`, `shamir`
+Manual locks: `invitation_creation`
+Special checks:
+
+- `last_vlob_timestamp`: Compare the to-be-inserted data's timestamp with the timestamp
+  of the most recent vlob.
+- `user_revoke_safe`: See [User revocation and concurrent operation involving users](#user-revocation-and-concurrent-operation-involving-users)
+
+<!-- See `misc/check_server_readme.py` script that checks this table is up to date ! -->
+<!-- start-table-what-is-lock-by-who -->
+API Family    |  command                               | common | realm | sequester | shamir | invitation_creation | user_revoke_safe | last_vlob_timestamp |
```

Comment:
Should also add the invitation row lock used by invitation cancel/complete/step etc.)